addons/sale_loyalty_delivery/tests/test_free_shipping_reward.py

```diff
@@ -357,3 +357,45 @@ def test_nothing_delivered_nothing_to_invoice(self):
         order.action_confirm()
         self.assertEqual(order.delivery_set, True)
         self.assertEqual(order.invoice_status, 'no')
+
+    def test_delivery_shant_count_toward_quantity_bought(self):
+
+        # Create promotion: 10% for everything
+        discount_program = self.env['loyalty.program'].create({
+            'name': '10 percent off order with min. 2 products',
+            'trigger': 'auto',
+            'program_type': 'promotion',
+            'applies_on': 'current',
+            'rule_ids': [(0, 0, {
+                'minimum_qty': 2,
+                'minimum_amount':0,
+            })],
+           'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount_mode': 'percent',
+                'discount': 10.0,
+                'discount_applicability': 'order',
+            })],
+        })
+
+        # Create an order including: product and delivery
+        order = self.empty_order
+        self.env['sale.order.line'].create({
+            'product_id': self.iPadMini.id,
+            'name': 'Large Cabinet',
+            'product_uom_qty': 1.0,
+            'order_id': order.id,
+        })
+        self.env['sale.order.line'].create({
+            'product_id': self.product_delivery_poste.id,
+            'name': 'Free delivery charges\nFree Shipping',
+            'product_uom_qty': 1.0,
+            'order_id': order.id,
+            'is_delivery': True,
+        })
+
+        # Calculate promotions
+        self._auto_rewards(order, discount_program)
+
+        # Make sure the promotion is NOT added
+        self.assertEqual(len(order.order_line.ids), 2, "Order should contains 2 lines: one product and one delivery. 3 lines mean that discount was wrongfully added.")
```

Comment:
Could you ensure the line length doesn't go above 100 chars? =]

Reply:
done